main/webapp/modules/core/scripts/views/data-table/data-table-view.js

```diff
@@ -524,6 +526,17 @@ DataTableView.prototype._onChangeGotoPage = function(elmt, evt) {
   this._showRows((gotoPageNumber - 1) * this._pageSize);
 };
 
+DataTableView.prototype._onKeyDownGotoPage = function(elmt, evt) {
+  var keyDownCode = event.which;
+  
+  if([38, 40].indexOf(keyDownCode) == -1) return;
+
+  if(keyDownCode == 38) this._onClickPreviousPage(elmt, evt);
```

Comment:
Nice.  It's a bit sluggish, and not quick firing like on the Mozilla Dev example URL I gave earlier, perhaps nothing you can do about that since we're evaluating a bunch on the key press now it seems?

Also, we need to check if currentPageNumber is already 1 ... and we press Down Arrow key...then DO NOTHING (I.E. do not onClickNextPage.  

Because now, if it's on 1 and I press Down Arrow key on my keyboard...it increments to 2, instead of staying on 1.
Same goes for if i press Up Arrow key and the currentPageNumber = Max page number, it should stay at Max page number and not increase or decrease.

Reply:
And...you need to flip the Up/Down arrow keypress to do the opposite... DOWN is go down the sheet.  Users will expect it to increment, and when you press UP arrow, users would expect it to scroll up on row/record index, or in other words, decrement the page number.

Reply:
Going to bed... will check on updates in morning.

Reply:
> Nice. It's a bit sluggish, and not quick firing like on the Mozilla Dev example URL I gave earlier, perhaps nothing you can do about that since we're evaluating a bunch on the key press now it seems?

It's because since we delete/rebuilt control with each change of page, I must resort to an unorthodox method. I wait 1,5 s. for the data-panel to be rebuilt. This is ok, but not more. I'm not proud of that, but I believe you bear the responsability to send me to those murky waters… I'm kidding, of course, but still, this arrow thing is a bit stretching our current table management.

> Also, we need to check if currentPageNumber is already 1 ... and we press Down Arrow key...then DO NOTHING (I.E. do not onClickNextPage).
> 
> Because now, if it's on 1 and I press Down Arrow key on my keyboard...it increments to 2, instead of staying on 1.
> Same goes for if I press Up Arrow key and the currentPageNumber == Max page number, it should stay at Max page number and not increase.

It is now fixed. It stops at bounds.

> And...you need to flip the Up/Down arrow keypress to do the opposite... DOWN is go down the sheet. Users will expect it to increment, and when you press UP arrow, users would expect it to scroll up on row/record index, or in other words, decrement the page number.

I understand that, BUT… just beside are the `<input type="number">` arrows. And they follow the human number convention, where higher # has a meaning. So, we are doing the contrary. This is weird. You think about it.

> Going to bed... will check on updates in morning.
> 
> We're so close to perfection now!!!

Well, I guess you will have fun testing that tomorrow!

Regards,
   Antoine